components/flex/typings.ts

```diff
@@ -0,0 +1,44 @@
+/**
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/LICENSE
+ */
+
+export type nzDirection = 'row' | 'row-reverse' | 'column' | 'column-reverse';
+
+export type nzJustify =
+  | 'flex-start'
+  | 'center'
+  | 'flex-end'
+  | 'space-between'
+  | 'space-around'
+  | 'space-evenly'
+  | 'start'
+  | 'end'
+  | 'right'
+  | 'left'
+  | 'stretch'
+  | 'normal';
+
+export type nzAlign =
+  | 'flex-start'
+  | 'center'
+  | 'flex-end'
+  | 'space-between'
+  | 'space-around'
+  | 'space-evenly'
+  | 'start'
+  | 'end'
+  | 'right'
+  | 'left'
+  | 'stretch'
+  | 'normal';
+
+export type nzGap = 'small' | 'middle' | 'large' | nzCustomGap;
+export type nzCustomGap = number;
+
+export type nzWrap = 'wrap' | 'nowrap' | 'wrap-reverse';
+
+export type nzFlex = `${nzFlexShrink} ${nzFlexGrow} ${nzFlexBasis}` | 'unset';
+export type nzFlexShrink = number;
+export type nzFlexGrow = number;
+export type nzFlexBasis = number | `${number}px` | `${number}rem` | `${number}%` | 'auto';
```

Comment:
> I don't think we need to enumerate css units because there are too many of them and they are updated all the time.
> 
> `${number}px` | `${number}rem` | `${number}%` `->` `${number}${string}`

I agree with you, as I have not used other css units myself, I assumed they were not useful. However, you are right that we should not restrict the users to these types.

Reply:
> Or we can reuse the types defined by [csstype](https://github.com/frenic/csstype), what do you think?

I appreciate your suggestion, but I am reluctant to use this package as it has not been updated since 2018. However, I find the idea interesting. I have noticed that ng-zorro currently does not have stylelint, and I think supporting linting for styles and strong style typings in the future might be a good idea.

Reply:
Let's change the type to `string`, because I noticed there are more cases here, such as `var(--x)`, `calc(x)`, etc.

Reply:
I agree with you, I did not consider those cases. Changing the type to string makes sense.